javascript/spec/test.js

```diff
@@ -0,0 +1,66 @@
+import { expect } from 'chai'
+import oo from '../object_oriented.js'
+import functional from '../functional.js'
+import imperative from '../imperative.js'
+
+const test = (input, output) => {
+  const paradigms = [
+    ['Object Oriented', oo]
+    , ['Functional', functional]
+    , ['Imperative', imperative]
+  ]
+
+  for (let i in paradigms) {
+    const [name, fn] = paradigms[i]
+    it(name, () => {
+      expect(fn(input)).to.equal(output)
+    })
+  }
+}
+
+const nest = (element) => {
```

Comment:
it's WIP, the input/output will be obtained by calling `fs` to obtain the contents of a named file from the `test_data` directory - that's just there so that the initial tests will fail. "nest" generates tests for nested tags.